### ISSUE TITLE
Add Virtual Hosts column to "Requested Files"

### ISSUE THREAD

**Issue by Thor77** [allinurl/goaccess#435]
It's very confusing to me to see multiple requests to `/` in the `Requested Files`-table, delivering different content, because `/` is available on multiple Virtual Hosts.
I would prefer a `Virtual Hosts`-column on `Requested Files` or another way to split these requests up.

**Comment by allinurl** [comment]
You can prepend the vhost to the requested files. Please take a look at the [**man page**](https://goaccess.io/man#examples), Virtual Hosts examples at the bottom of the page.

The output would look like:

![2016-06-11-091951_1241x1054_scrot](https://cloud.githubusercontent.com/assets/5005367/15985525/ea5f9a26-2fb5-11e6-9051-8f388bd5a6d5.png)

**Comment by Thor77** [comment]
Yeah, I already found that option in an issue, but i don't really like the view (vhost + request in one column) of that.

**Comment by allinurl** [comment]
I can look into the new column. However, for the current output, and if it's just a spacing issue, you can just add x amount of spaces between the two awk fields, e.g., `'$8=$1" "$8'`.

**Comment by uimanwashere** [comment]
Also it seems that appending the hostname with awk disables the real-time preview, right?. In that case, creating a column for %v could solve this issue.

**Comment by allinurl** [comment]
@uimanwashere you are correct. I'll add this option. Thanks.

**Comment by virtualdreams** [comment]
Please add this column also to Static Requests, Not Found URLs.

**Comment by fancsali** [comment]
@allinurl I just bumped into the same problem, and then found this issue. I was just wondering, whether this is still on the roadmap?

**Comment by fkrauthan** [comment]
Yeah in general most of the panels would be super helpful to have virtual host as a separate column. In particular

- Requested files
- Static requests
- Not found URLs

For most of the other panels I think the filtering that is currently planned (as part of #117) Is enough. Even though a groupBy for the other panels rather then filtering could be super helpful if I want to see a quick health check e.g. which of my vhosts throws the most 5xx errors etc.

**Comment by allinurl** [comment]
@fkrauthan I agree that #117 should make it possible to say, for example, "show me all the 500x errors for this vhost." At the same time, I understand the value in being able to view data for a specific vhost while displaying multiple vhosts on the dashboard. Adding the virtual host in those three panels makes sense, especially when you want to compare several vhosts in a single view. Maybe it could be added as an optional column, hidden by default?

That said, there's also the challenge of how to report it. I assume a single vhost can have multiple METHOD/URI/PROTOCOL combinations, which could result in a long list of entries for just one vhost. Thoughts?

**Comment by fkrauthan** [comment]
I guess the real question is how valuable is it to not co-relate the information with vHost in particular. /home for wordpress has a completely different meaning to me as /home of my personal website for example. So I think showing more lines is not the end of the world. Especially with filters mixed in this should be even less an issue.

**Comment by mk12** [comment]
I found it very confusing that goaccess parses virtual hosts (and even has a `--fname-as-vhost` flag) but does not show the domain, except in the Virtual Hosts panel. When you have multiple virtual hosts, just looking at the path is meaningless.

For those using Caddy, here's a workaround to show the virtual host:

```
jq -c '.request |= (.uri = .host + .uri)' /var/log/caddy/access.log | goaccess --log-format CADDY
```

**Comment by sokai** [comment]
Hey everyone!

**At OP:** May I ask to have the requested (host/vhost) column to (the tables) `Static Requests` and `Not Found URLs (404s)`, also?

@mk12: Thanks for the suggestion!
> For those using Caddy, here's a workaround to show the virtual host:
> `jq -c '.request |= (.uri = .host + .uri)' /var/log/caddy/access.log | goaccess --log-format CADDY`

I tried this but can't _experience any impact_ at my current real-time HTML report w/ (`zgrep` and) multiple Caddy vhosts …
Is `goaccess` cutting off the prepended vhost from the `request.uri` field _somehow/anyway_, is it working at your setup? 🤔

JSON data (from Caddy logs) **w/ _workaround_**:
```json

{
  "level": "info",
  […]
  "request": {
    […]
    "host": "my.host.tld",
    "uri": "my.host.tld/file.ext",
    […]
  }
}
```

JSON data (from Caddy logs) **w/o _workaround_**:
```json
{
  "level": "info",
  […]
  "request": {
    […]
    "host": "my.host.tld",
    "uri": "/file.ext",
    […]
  }
}
```

Thanks and KR

**Comment by mk12** [comment]
@sokai The command I gave is just for processing the logs one time. For the live HTML view you'd need to do it continuously:

```
tail -f -n+1 /var/log/caddy/access.log | jq -c --unbuffered '.request |= (.uri = .host + .uri)' | goaccess --log-format CADDY --real-time-html -o index.html -
```

**Comment by mk12** [comment]
Although I can confirm that when I run the live HTML view that way, only some of the paths have a host attached, not all. Will need to investigate further what’s going on.
